Sources/BitcoinKit/Messages/BlockMessage.swift

```diff
@@ -77,4 +77,31 @@ public struct BlockMessage {
         }
         return BlockMessage(version: version, prevBlock: prevBlock, merkleRoot: merkleRoot, timestamp: timestamp, bits: bits, nonce: nonce, transactionCount: transactionCount, transactions: transactions)
     }
+
+    public func computeMerkleRoot() -> Data {
+        var hashes = transactions.map { (tx) -> Data in tx.txHash }
+
+        while hashes.count > 1 {
+            if hashes.count % 2 != 0 {
+                let last = hashes.last!
+                hashes.append(last)
+            }
+
+            var i = 0
```

Comment:
Using `for loop` make codes shorter instead of `while loop`.

```
for i in 0..<hashes.count / 2 {
    let left = hashes[2 * i]
    let right = hashes[2 * i + 1]

    var parent = Data()
    parent.append(left)
    parent.append(right)

    hashes[i] = Crypto.sha256sha256(parent)
}
```